tests/DependencyInjection/Compiler/AdminSearchCompilerPassTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Tests\DependencyInjection\Compiler;
+
+use PHPUnit\Framework\TestCase;
+use Sonata\AdminBundle\DependencyInjection\Compiler\AdminSearchCompilerPass;
+use Sonata\AdminBundle\Tests\Fixtures\Admin\PostAdmin;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+/**
+ * @author Javier Spagnoletti <phansys@gmail.com>
+ */
+final class AdminSearchCompilerPassTest extends TestCase
```

Comment:
What about using [`AbstractCompilerPassTestCase`](https://github.com/SymfonyTest/SymfonyDependencyInjectionTest#testing-a-compiler-pass)?

Reply:
Sorry, I thought I had written something more in the previous comment. What I meant was to use that class to make the test simpler and to use real objects, something like:

```php
$adminFooDefinition = new Definition(PostAdmin::class);
$adminFooDefinition->addTag('sonata.admin', ['global_search' => true]);
$this->setDefinition('admin.foo', $adminFooDefinition);

$adminBarDefinition = new Definition(PostAdmin::class);
$adminBarDefinition->addTag('sonata.admin', ['global_search' => false]);
$this->setDefinition('admin.bar', $adminBarDefinition);

$adminBazDefinition = new Definition(PostAdmin::class);
$adminBazDefinition->addTag('sonata.admin', ['some_attribute' => 42]);
$this->setDefinition('admin.baz', $adminBazDefinition);

$searchHandlerDefinition = new Definition();
$this->setDefinition('sonata.admin.search.handler', $searchHandlerDefinition);

$this->compile();

$this->assertContainerBuilderHasServiceDefinitionWithMethodCall(
    'sonata.admin.search.handler',
    'configureAdminSearch',
    [['admin.foo' => true, 'admin.bar' => false]]
);
```

Reply:
Thank you. I've applied your suggestion.